### PR TITLE
refactor(components): update checkbox styling with re-theme flag [JOB-84551]

### PR DESCRIPTION
**Pull request by chris-at-jobber**
## Motivations

Allow checkbox to be re-styled when a given class is present

### Changed

- Enable checkbox to be re-themed and create alternate styling for re-theme

## Testing

Apply the `jobber-retheme` class to the Storybook iframe `body`

- [ ] With `jobber-retheme`, should look different
- [ ] Without `jobber-retheme`, should look the same

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/Checkbox/Checkbox.css [hunk 1]
```diff
@@ -45,6 +50,15 @@
   border-color: var(--color-disabled--secondary);
 }
 
+:global(.jobber-retheme) .checkBox {
+  border-color: var(--color-border);
+  border-radius: calc(var(--radius-base) / 2);
+}
+
+:global(.jobber-retheme) .checkBox:hover {
```
Comment (fagnanm) [inline packages/components/src/Checkbox/Checkbox.css:58]: @chris-at-jobber can we apply this to the label as well

File: packages/components/src/Checkbox/Checkbox.css [hunk 2]
```diff
@@ -45,6 +50,15 @@
   border-color: var(--color-disabled--secondary);
 }
 
+:global(.jobber-retheme) .checkBox {
+  border-color: var(--color-border);
+  border-radius: calc(var(--radius-base) / 2);
```
Comment (fagnanm) [inline packages/components/src/Checkbox/Checkbox.css:55]: @chris-at-jobber should we do this? Or just set to radius-small and then bump small up to 4 and base up to 8?

**Review by fagnanm — CHANGES_REQUESTED** [review]
Mostly looks good, one question and an update to make so that the hover state applies to labels as well

File: packages/components/src/Checkbox/Checkbox.css (hunk 2 above)
Comment (chris-at-jobber) [reply packages/components/src/Checkbox/Checkbox.css:55]: Yeah that's cleaner, I'll do that

Comment (chris-at-jobber) [reply packages/components/src/Checkbox/Checkbox.css:55]: Just realized I also need to update the design package to add a radius-small but we need to do that anyways so why not now 😄

**Comment by chris-at-jobber** [comment]
@fagnanm radius and hover state updated

**Review by fagnanm — APPROVED** [review]
Changes look good. TIL about focus-visible!

**Pull request merged**